scalar/scalar.go

```diff
@@ -0,0 +1,30 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package scalar
+
+// Scalar represent a set of useful types to be used with type parameters.
+//
+// For now we only support the type that are frequently used within the
+// controller repositories, code-generator and runtime.
+type Scalar interface {
+	int | int32 | int64 | float32 | float64 | bool | string
+}
```

Comment:
After looking at these methods, and what `lo` supports, I don't think we ever need to constrain by `Scalar`. If `lo` doesn't support the generic, we should make our own method that supports a more generic constraint like `comparable` or even `any`.

Reply:
The reason why I thought about bringing this package is more to get rid of `equalString`, `equalInt` ... that exist in most of our hook.go files. It is definitely extendable, maybe let's start with a simple prototype and see what we can add later on?